akka-actor/src/main/scala/akka/pattern/RetrySupport.scala

```diff
@@ -251,6 +249,17 @@ trait RetrySupport {
       delayFunction: Int => Option[FiniteDuration])(implicit ec: ExecutionContext, scheduler: Scheduler): Future[T] = {
     RetrySupport.retry(attempt, attempts, delayFunction, attempted = 0, shouldRetry)
   }
+
+  def retry[T](attempt: () => Future[T], retrySettings: RetrySettings)(
+      implicit ec: ExecutionContext,
+      scheduler: Scheduler): Future[T] = {
+    RetrySupport.retry(
+      attempt,
+      retrySettings.maxRetries,
+      retrySettings.delayFunction,
+      attempted = 0,
+      retrySettings.shouldRetry)
+  }
```

Comment:
I don't remember, do we have something clever that will get classic scheduler from a typed actor system? If not we'd want a method with implicit `ClassicActorSystemProvider` so it can be used for typed as well (can probably not be an overload, but just having the provider version seems more convenient anyway).

Reply:
a method with the implicit `ClassicActorSystemProvider` was added to `Patterns`, here I don't see any existing method with that, do you think it's worth adding it?

Reply:
If we don't it won't be easy to use it from Scala based Akka typed apps, so better have that than the existing one (note that there is one classic Scheduler and one typed).